IS/Sem.07/Pract.07/References/Task04.cpp

```diff
@@ -0,0 +1,46 @@
+#include <iostream>
+using namespace std;
+void DenominatorAndNumerator(int& num1, int& num2);
+void Sum(int& num1, int& num2, int& num3, int& num4);
+
+int main()
+{
+    int num1, num2, num3, num4;
+	cin >> num1 >> num2 >> num3 >> num4;
+	DenominatorAndNumerator(num1, num2);
+	DenominatorAndNumerator(num3, num4);
+	Sum(num1, num2, num3, num4);
+}
+
+void DenominatorAndNumerator(int& num1, int& num2)
+{
+	int helper = (num1 >= num2) ? num2 : num1;
+	while (num1 > helper / 2 || num2 > helper / 2)
+	{
+		for (int i = 2; i <= helper / 2; i++)
+		{
+			if (num1 % i == 0 && num2 % i == 0)
+			{
+				num1 /= i;
+				num2 /= i;
+			}
+		}
+	}
+}
```

Comment:
Infinite loop, rethink you logic, try to do it without the helper and try to do it with gcd or lcm(lowest common divisor)
Try input 2 2 3 3